homeassistant/components/syncthru/binary_sensor.py

```diff
@@ -0,0 +1,107 @@
+"""Support for Samsung Printers with SyncThru web interface."""
+
+import logging
+
+from homeassistant.components.binary_sensor import (DEVICE_CLASS_CONNECTIVITY,
+                                                    DEVICE_CLASS_PROBLEM,
+                                                    BinarySensorEntity)
+from homeassistant.const import CONF_NAME
+from pysyncthru import SyncThru, SyncthruState
+
+from . import device_identifiers
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SYNCTHRU_STATE_PROBLEM = {
+    SyncthruState.INVALID: True,
+    SyncthruState.OFFLINE: None,
+    SyncthruState.NORMAL: False,
+    SyncthruState.UNKNOWN: True,
+    SyncthruState.WARNING: True,
+    SyncthruState.TESTING: False,
+    SyncthruState.ERROR: True,
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up from config entry."""
+
+    printer = hass.data[DOMAIN][config_entry.entry_id]
+
+    name = config_entry.data[CONF_NAME]
+    devices = [
+        SyncThruOnlineSensor(printer, name),
+        SyncThruProblemSensor(printer, name),
+    ]
+
+    async_add_entities(devices, True)
+
+
+class SyncThruBinarySensor(BinarySensorEntity):
+    """Implementation of an abstract Samsung Printer binary sensor platform."""
+
+    def __init__(self, syncthru, name):
+        """Initialize the sensor."""
+        self.syncthru: SyncThru = syncthru
+        self._state = None
+        self._name = name
+        self._id_suffix = ""
+
+    @property
+    def unique_id(self):
+        """Return unique ID for the sensor."""
+        serial = self.syncthru.serial_number()
+        return serial + self._id_suffix if serial else super().unique_id
```

Comment:
Please use an f-string instead of string concatenation.
```suggestion
        return f"{serial}{self._id_suffix}" if serial else None
```